### ISSUE TITLE
polymer 2 - build fails

### ISSUE THREAD

**Issue by madhusudhand** [Polymer/polymer-cli#669]
## Description
the command `polymer build --entrypoint elements.html` fails with the following errors.

**elements.html** contains
```
<link href="../../bower_components/paper-button/paper-button.html" rel="import">
```

`error`:   Promise rejection: Error: Error parsing script in bower_components/polymer/lib/legacy/class.html at NaN:NaN
TypeError: Cannot read property 'forEach' of undefined
    at Object.enterMethodDefinition (/Users/madhusudhanad/ng4/angular4-polymer/node_modules/hydrolysis/lib/ast-utils/element-finder.js:92:38)
    at applyVisitors (/Users/madhusudhanad/ng4/angular4-polymer/node_modules/hydrolysis/lib/ast-utils/js-parse.js:31:51)
    at Controller.enter (/Users/madhusudhanad/ng4/angular4-polymer/node_modules/hydrolysis/lib/ast-utils/js-parse.js:54:20)
    at Controller.__execute (/Users/madhusudhanad/ng4/angular4-polymer/node_modules/estraverse/estraverse.js:395:31)
    at Controller.traverse (/Users/madhusudhanad/ng4/angular4-polymer/node_modules/estraverse/estraverse.js:493:28)
    at Object.traverse (/Users/madhusudhanad/ng4/angular4-polymer/node_modules/estraverse/estraverse.js:705:27)
    at Object.jsParse (/Users/madhusudhanad/ng4/angular4-polymer/node_modules/hydrolysis/lib/ast-utils/js-parse.js:75:16)
    at Analyzer._processScript (/Users/madhusudhanad/ng4/angular4-polymer/node_modules/hydrolysis/lib/analyzer.js:318:43)
    at /Users/madhusudhanad/ng4/angular4-polymer/node_modules/hydrolysis/lib/analyzer.js:301:44
    at Array.forEach (native)
error:   Error: Error parsing script in bower_components/polymer/lib/legacy/class.html at NaN:NaN
TypeError: Cannot read property 'forEach' of undefined
    at Object.enterMethodDefinition (/Users/madhusudhanad/ng4/angular4-polymer/node_modules/hydrolysis/lib/ast-utils/element-finder.js:92:38)
    at applyVisitors (/Users/madhusudhanad/ng4/angular4-polymer/node_modules/hydrolysis/lib/ast-utils/js-parse.js:31:51)
    at Controller.enter (/Users/madhusudhanad/ng4/angular4-polymer/node_modules/hydrolysis/lib/ast-utils/js-parse.js:54:20)
    at Controller.__execute (/Users/madhusudhanad/ng4/angular4-polymer/node_modules/estraverse/estraverse.js:395:31)
    at Controller.traverse (/Users/madhusudhanad/ng4/angular4-polymer/node_modules/estraverse/estraverse.js:493:28)
    at Object.traverse (/Users/madhusudhanad/ng4/angular4-polymer/node_modules/estraverse/estraverse.js:705:27)
    at Object.jsParse (/Users/madhusudhanad/ng4/angular4-polymer/node_modules/hydrolysis/lib/ast-utils/js-parse.js:75:16)
    at Analyzer._processScript (/Users/madhusudhanad/ng4/angular4-polymer/node_modules/hydrolysis/lib/analyzer.js:318:43)
    at /Users/madhusudhanad/ng4/angular4-polymer/node_modules/hydrolysis/lib/analyzer.js:301:44
    at Array.forEach (native)
    at Analyzer._processScript (/Users/madhusudhanad/ng4/angular4-polymer/node_modules/hydrolysis/lib/analyzer.js:331:36)
    at /Users/madhusudhanad/ng4/angular4-polymer/node_modules/hydrolysis/lib/analyzer.js:301:44
    at Array.forEach (native)
    at Analyzer._processScripts (/Users/madhusudhanad/ng4/angular4-polymer/node_modules/hydrolysis/lib/analyzer.js:300:21)
    at Analyzer._parseHTML (/Users/madhusudhanad/ng4/angular4-polymer/node_modules/hydrolysis/lib/analyzer.js:161:39)
    at Timeout._onTimeout (/Users/madhusudhanad/ng4/angular4-polymer/node_modules/hydrolysis/lib/analyzer.js:125:39)
    at ontimeout (timers.js:365:14)
    at tryOnTimeout (timers.js:237:5)
    at Timer.listOnTimeout (timers.js:207:5)

### Versions & Environment

- Polymer CLI: 0.17.0
- node: 6.9.2
- Operating System: MacOS Sierra

#### Steps to Reproduce

1. Create an angular project (ng new project-name).
2. Create a elements.html inside assets and ad an entry.

<link href="../../bower_components/paper-button/paper-button.html" rel="import">

3. Build: `polymer build --entrypoint src/assets/elements.html`

**Comment by jsilvermist** [comment]
Is this problem fixed if you update `polymer-cli` to `0.18.0`?

**Comment by madhusudhand** [comment]
Yes
